logic/src/commonMain/kotlin/com/wire/kalium/logic/data/user/UserMapper.kt

```diff
@@ -2,36 +2,88 @@ package com.wire.kalium.logic.data.user
-    fun fromApiModelToDaoModel(selfUserInfoResponse: SelfUserInfoResponse): PersistedUser
-    fun fromDaoModel(user: PersistedUser): SelfUser
+    fun fromApiModelToDaoModel(userDetailsResponse: UserDetailsResponse): UserEntity
+    fun fromApiModelToDaoModel(selfUserInfoResponse: SelfUserInfoResponse): UserEntity
+    fun fromDaoModel(user: UserEntity): SelfUser
+
+    /**
+     * Maps the user data to be updated. if the parameters [newName] [newAccent] [newAssetId] are nulls,
+     * it indicates that not updation should be made.
+     *
+     *  TODO: handle deletion of assets references, emptyAssetList
+     */
+    fun fromModelToUpdateApiModel(user: SelfUser, newName: String?, newAccent: Int?, newAssetId: String?): UserUpdateRequest
 }
 
 internal class UserMapperImpl(private val idMapper: IdMapper) : UserMapper {
 
     override fun fromApiModel(selfUserInfoResponse: SelfUserInfoResponse): SelfUser {
-        return SelfUser(idMapper.fromApiModel(selfUserInfoResponse.qualifiedId))
+        return SelfUser(
+            idMapper.fromApiModel(selfUserInfoResponse.qualifiedId),
+            selfUserInfoResponse.name,
+            selfUserInfoResponse.handle,
+            selfUserInfoResponse.email,
+            selfUserInfoResponse.phone,
+            selfUserInfoResponse.accentId,
+            selfUserInfoResponse.team,
+            emptyList()
+        )
     }
 
-    override fun fromApiModelToDaoModel(userDetailsResponse: UserDetailsResponse): PersistedUser {
-        return PersistedUser(idMapper.fromApiToDao(userDetailsResponse.id), userDetailsResponse.name, userDetailsResponse.handle)
+    override fun fromApiModelToDaoModel(userDetailsResponse: UserDetailsResponse): UserEntity {
+        return UserEntity(
+            idMapper.fromApiToDao(userDetailsResponse.id),
+            userDetailsResponse.name,
+            userDetailsResponse.handle,
+            null,
+            null,
+            userDetailsResponse.accentId,
+            null
+        )
     }
 
-    override fun fromDaoModel(user: com.wire.kalium.persistence.dao.User): SelfUser {
-        return SelfUser(idMapper.fromDaoModel(user.id))
+    override fun fromDaoModel(user: com.wire.kalium.persistence.dao.UserEntity) =
+        SelfUser(idMapper.fromDaoModel(user.id), user.name, user.handle, user.email, user.phone, user.accentId, user.team, emptyList())
+
+    override fun fromModelToUpdateApiModel(
+        user: SelfUser,
+        newName: String?,
+        newAccent: Int?,
+        newAssetId: String?
+    ): UserUpdateRequest {
+        return UserUpdateRequest(
+            id = user.id.value,
+            qualifiedId = idMapper.toApiModel(user.id),
+            name = newName ?: user.name,
+            accentId = newAccent ?: user.accentId,
+            assets = if (newAssetId != null) {
+                listOf(
+                    UserAssetRequest(newAssetId, ImageSize.Complete),
+                    UserAssetRequest(newAssetId, ImageSize.Preview)
```

Comment:
Missing TODO?

The preview image should a separate downscaled image of the complete image.

Reply:
This part I had some doubts, specially looking at the current app code and having previews for the avatar or not. 
I will revisit on the next PR when we will be saving these assets into the upcoming `Assets` table for sure, at least this enables us to start working with the profile picture from the Reloaded app =)